src/mongocxx/options/find.hpp

```diff
@@ -26,13 +26,14 @@ namespace mongocxx {
 MONGOCXX_INLINE_NAMESPACE_BEGIN
 namespace options {
 
+enum class cursor_type : std::uint8_t { k_non_tailable, k_tailable, k_tailable_await };
```

Comment:
a liberal sprinkling of typename does the trick:

https://github.com/hanumantmk/mongo-cxx-driver/commit/2719bc0ed6a5462f97182b3d466acfc47cfc9ce9